CAMEL-12462

davsclaus opened the pull request:
See ticket: https://issues.apache.org/jira/browse/CAMEL-12462

Anyone is welcome to take a peak and provide feedback. This PR can be squashed and merged when its done.

We can do this kind of optimisation for other components as well, for example JMS with the CamelJmsDestination header for the JMS queue/topic. However JMS uses a connection pool so its fairly optimised as well, but it would still use a new producer per different queue/topic.

fharms commented:
Nice. I really like this feature. 

How difficult would it be to also make camel-jetty support this optimization ?

davsclaus commented:
The camel-jetty producer is deprecated (not the jetty consumer) - so consider switching to another component for http. But I do think its possible to add support for it, let me take a look.

fharms commented:
Ohh I missed that. No need for adding support for it then, I just switch to another http component.

davsclaus commented:
Okay added support for jetty as well, as it was easy. Updated the toD docs a bit more about cache size is not your solution, and added a solution with static `to`.

davsclaus commented on components/camel-http/src/test/java/org/apache/camel/component/http/HttpSendDynamicAwareTest.java:
```diff
@@ -0,0 +1,96 @@
+                from("direct:joes")
+                    .toD("http://localhost:" + PORT + "/joes?throwExceptionOnFailure=false&drink=${header.drink}");
+
+                from("direct:vokda")
```
Ah sorry lets get that fixed
